RecordType does not implement Deserialize... but it does?

**Describe the bug**
Trying to just include the `RecordType` in a struct which derives `Deserialize` and it's not working

**To Reproduce**
code:

```rust
// main.rs
use serde::Deserialize;
use trust_dns_client::rr::RecordType;

#[derive(Deserialize)]
struct Test {
    test: RecordType,
}

fn main() {
    println!("Hello, world!");
}
```

```toml
# Cargo.toml
[package]
name = "test"
version = "0.1.0"
edition = "2021"

[dependencies]
trust-dns-client = {version = "0.23", features = ["serde-config"]}
serde = {version = "*", features = ["derive"]}
```

**Expected behavior**
`RecordType` is compiled with `Deserialize` due to the `serde-config` feature being used

**Actual behavior**
```
error[E0277]: the trait bound `RecordType: Deserialize<'_>` is not satisfied
    --> src/main.rs:6:11
     |
6    |     test: RecordType,
     |           ^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `RecordType`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a str
               ()
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 129 others
note: required by a bound in `next_element`
    --> /Users/redacted/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.188/src/de/mod.rs:1724:12
     |
1724 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `RecordType: Deserialize<'_>` is not satisfied
    --> src/main.rs:6:11
     |
6    |     test: RecordType,
     |           ^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `RecordType`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a str
               ()
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 129 others
note: required by a bound in `next_value`
    --> /Users/redacted/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.188/src/de/mod.rs:1863:12
     |
1863 |         V: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `MapAccess::next_value`

error[E0277]: the trait bound `RecordType: Deserialize<'_>` is not satisfied
  --> src/main.rs:6:5
   |
6  |     test: RecordType,
   |     ^^^^^^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `RecordType`
   |
   = help: the following other types implement trait `Deserialize<'de>`:
             &'a Path
             &'a [u8]
             &'a str
             ()
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
             (T0, T1, T2, T3, T4)
           and 129 others
note: required by a bound in `_serde::__private::de::missing_field`
  --> /Users/redacted/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.188/src/private/de.rs:25:8
   |
25 |     V: Deserialize<'de>,
   |        ^^^^^^^^^^^^^^^^ required by this bound in `missing_field`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `test` (bin "test") due to 3 previous errors
```

**System:**
 - OS: macos
 - Architecture: arm64
 - Version 13.5
 - rustc version: 1.71 nightly

**Version:**
Crate: trusty-dns-client
Version: 0.23

--- a/crates/client/Cargo.toml
+++ b/crates/client/Cargo.toml
@@ -55,7 +55,7 @@ dnssec-openssl = ["dnssec", "trust-dns-proto/dnssec-openssl"]
 dnssec-ring = ["dnssec", "trust-dns-proto/dnssec-ring"]
 dnssec = ["trust-dns-proto/dnssec"]
 
-serde-config = ["serde"]
+serde-config = ["serde", "trust-dns-proto/serde-config"]
 
 # enables experimental the mDNS (multicast) feature
 mdns = ["trust-dns-proto/mdns"]

